ui_curses.c

```diff
@@ -1624,11 +1624,11 @@ int yes_no_query(const char *format, ...)
 	while (1) {
 		int ch = getch();
 
-		if (ch == ERR || ch == 0)
-			continue;
-		if (ch == 'y')
+		if (ch == 'y' || !cmus_running) {
```

Comment:
why do we treat `!cmus_running` as "yes"?

Reply:
Well !cmus_running means that  cmus should exit, because it received a SIGTERM for example or something along the lines. If we treat it as 'n' cmus gets zombified and continues playing in the background.

Reply:
if we ignore cmus_running altogether it, cmus falls in an endless loop with 100% cpu utilisation under the conditions described in #765

Reply:
`yes_no_query` can be called in different places, and "no" is a better default answer (in general). Imagine if `rm -rf /` when killed while prompting "are you sure" would treat that as "yes" :-)

This fix looks more like a workaround than a solution.

Reply:
yes you are right in part. But returning "no" is also not an option. As is cmus zombifies if we return "no".

To fix this issue we will have to refactor the API to:

```bool yes_no_query(const char const * msg, char * res)```

And then change all invocations to check if the function for success before using the result. I could do that, but I'd prefer if we could work this out ahead of time, so we don't have that much back and forth.

Reply:
Sorry the function signature above is total crap.
Point is we need three return values: yes, no, error

Reply:
Maybe return an enum and check that?

Reply:
Sure, I have already started making the necessary changes.